str0m/src/stats.rs

```diff
@@ -0,0 +1,147 @@
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::VecDeque,
+    time::{Duration, Instant},
+};
+
+use crate::{Mid, Rtc};
+use rtp::Rid;
+
+pub struct Stats {
+    last_snapshot: StatsSnapshot,
+    events: VecDeque<PeerStats>,
+}
+
+type Bytes = u64;
+
+pub struct StatsSnapshot {
+    pub peer_tx: Bytes,
+    pub peer_rx: Bytes,
+    pub tx: Bytes,
+    pub rx: Bytes,
+    pub ts: Instant,
+}
+
+impl StatsSnapshot {
+    fn new(ts: Instant) -> StatsSnapshot {
+        // TODO: is there a way to only provide the `ts` property
+        // which does not conform to Default,
+        // and have the rest generated by ..Default::default() ?
```

Comment:
Unfortunately not, I think I recall seeing an RFC to address this but I can't find it

Reply:
amazing, thx